near-sdk-macros/src/core_impl/info_extractor/visitor.rs

```diff
@@ -209,26 +220,26 @@ fn is_view(sig: &Signature) -> bool {
     }
 }
 
-fn parse_return_kind(typ: &Type, handles_result: ResultHandling) -> syn::Result<ReturnKind> {
+fn parse_return_kind(
+    typ: &Type,
+    handles_result: ResultHandling,
+    persist_on_error: bool,
+) -> syn::Result<ReturnKind> {
     match handles_result {
-        ResultHandling::NoCheck => Ok(ReturnKind::HandlesResult(typ.clone())),
+        ResultHandling::NoCheck => Ok(ReturnKind::HandlesResultExplicit(typ.clone())),
         ResultHandling::Check => {
             if !utils::type_is_result(typ) {
                 Err(Error::new(typ.span(), "Function marked with #[handle_result] should return Result<T, E> (where E implements FunctionError). If you're trying to use a type alias for `Result`, try `#[handle_result(aliased)]`."))
             } else {
-                Ok(ReturnKind::HandlesResult(typ.clone()))
+                Ok(ReturnKind::HandlesResultExplicit(typ.clone()))
             }
         }
         ResultHandling::None => {
             if utils::type_is_result(typ) {
```

Comment:
It seems that you rely on this function, but this means that we won't be able to support Result aliases that are [not named `Result`](https://github.com/near/near-sdk-rs/blob/e22f74ea9f6a0be8fe030a9225716eab811116ea/near-sdk-macros/src/core_impl/utils/mod.rs#L10-L25).

There was an effort to express it with the type system: https://github.com/near/near-sdk-rs/pull/1086